packages/vite/src/node/plugins/worker.ts

```diff
@@ -226,8 +223,10 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
 
     async transform(raw, id) {
       const query = parseRequest(id)
-      if (query && query[WORKER_FILE_ID] != null && query['type'] != null) {
-        const workerType = query['type'] as WorkerType
+      if (query && query[WORKER_FILE_ID] != null) {
+        // if import worker by worker constructor will had query.type
+        // other type will be import worker by esm
+        const workerType = (query['type'] || 'module') as WorkerType
```

Comment:
Hmm I thought the default is to fallback to `classic`? 🤔

Reply:
in module query
```js
import worker from "./worker.js?worker"
```

will generate code

```js
export default function WorkerWrapper() {
  return new worker("./worker.js?worker_file", { type: 'module' })
}
```

in worker constructor

```js
new Worker(new URL("./worker.js"))
```

will generate code

```js
new Worker("./worker.js?worker_file&type=classic")
```

so if type no value it would be `module` 😁

Reply:
Oh that's interesting. Would it make sense to have the first code example generate code with `classic` instead? Since `module` workers only seem to be supported by Chrome only.

If not we should also update the [docs](https://vitejs.dev/guide/features.html#import-with-constructors). The second code example:

> The worker constructor also accepts options, which can be used to create "module" workers:

Implies that it's `classic` by default, but we're changing to `module` by default.

Reply:
Yes, `classic` should be the default. We may have changed it by mistake at one point.

Reply:
The first code now is no way to generate `classic worker`, and we suggest the user use the second way the generate `classic worker`?

in-browser `new Worker(new URL("./worker.js"))` is default to create classic worker. the type query params just call vite how to import `/@vite/env`.  And the second code always had the type query.

Reply:
It mean we should change `import worker from "./worker?worker"` to classic worker?

Reply:
> 

Oh, I know. the import worker will packaging by `iife`, import expression will clean. And we should new a classic worker in build mode 🤦